stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py

```diff
@@ -668,12 +673,22 @@ async def get_all_catalogs(
         )
 
         hits = response["hits"]["hits"]
-        catalogs = [
-            self.catalog_serializer.db_to_stac(
-                catalog=hit["_source"], base_url=base_url
+        catalogs = []
+        for hit in hits:
+            collections, _ = await self.get_catalog_collections(
```

Comment:
rather than calling await in a loop you can call `await asyncio.gather`
```python
results = await asyncio.gather([
    catalog_ids=[hit["_source"].get("id")],
    base_url=base_url,
    limit=100,
    token=None,
) for hit in hits], return_exceptions=True)

for result in results:
    if not isinstance(result, Exception):
        catalogs.append(
            self.catalog_serializer.db_to_stac(
            catalog=hit["_source"],
            base_url=base_url,
            collections=collections,
            conformance_classes=conformance_classes,
        )
)
```

This way you call all the async methods at the same time rather than waiting for each to return, which will improve speed, especially if you have a lot of hits.

Reply:
Good suggestion, was not aware of this function. I have updated as suggested.